### PR TITLE
[300] create unit test to package mapping

### PR DESCRIPTION
**Pull request by RockTF**
Create unit test to GreenCityMVP service/src/main/java/greencity/mapping

 CategoryDtoMapper
 CategoryDtoResponseMapper
 CustomHabitMapper
 CustomShoppingListMapper
 CustomShoppingListResponseDtoMapper
 EcoNewsAuthorDtoMapper
 EcoNewsCommentDtoMapper
 EcoNewsCommentVOMapper
 EcoNewsDtoMapper
 EcoNewsVOMapper
 FilterDtoRequestMapper
 FilterDtoResponseMapper
 HabitAssignDtoMapper
 HabitAssignManagementDtoMapper
 HabitAssignMapper
 HabitAssignUserDurationDtoMapper
 HabitDtoMapper
 HabitFactDtoResponseMapper
 HabitManagementDtoMapper
 HabitStatisticDtoMapper
 HabitStatusCalendarMapper
 HabitStatusCalendarVOMapper
 HabitTranslationDtoMapper
 HabitTranslationMapper
 LanguageTranslationDtoMapper
 MultipartBase64ImageMapper
 NewTagDtoMapper
 SearchNewsDtoMapper
 ShoppingListItemDtoMapper
 ShoppingListItemRequestDtoMapper
 ShoppingListItemResponseDtoMapper
 ShoppingListItemWithStatusRequestDtoMapper
 TagDtoMapper
 TagMapper
 TagVOMapper
 UtilsMapper

### CODE REVIEW

File: service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+@ExtendWith(MockitoExtension.class)
+class CategoryDtoResponseMapperTest {
+
+    @InjectMocks
+    CategoryDtoResponseMapper mapper;
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java]: should be private

File: service/src/test/java/greencity/mapping/EcoNewsCommentDtoMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
+class EcoNewsCommentDtoMapperTest {
+
+    @InjectMocks
+    private EcoNewsCommentDtoMapper mapper;
+
+    @Test
+    void convert() {
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/EcoNewsCommentDtoMapperTest.java:28]: Display Name annotation is missing

File: service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class HabitTranslationDtoMapperTest {
+
+    HabitTranslationDtoMapper habitTranslationDtoMapper = new HabitTranslationDtoMapper();
+
+    HabitTranslation habitTranslation;
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java:16]: private is missing

File: service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                .name(habitTranslation.getName())
+                .languageCode(habitTranslation.getLanguage().getCode())
+                .habitItem(habitTranslation.getHabitItem())
+                .description(habitTranslation.getDescription())
+                .build();
+
+    }
+
+    @Test
+    public void testConvert() {
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java]: Display Name is missing

File: service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class HabitTranslationMapperTest {
+
+    HabitTranslationMapper habitTranslationMapper = new HabitTranslationMapper();
+
+    HabitTranslation habitTranslation;
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java:16]: private is missing

File: service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+        habitTranslation = HabitTranslation.builder()
+                .name(habitTranslationDto.getName())
+                .description(habitTranslationDto.getDescription())
+                .habitItem(habitTranslationDto.getHabitItem())
+                .build();
+    }
+
+    @Test
+    public void testConvert() {
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java]: display name is missing

File: service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        Assertions.assertEquals(habitTranslation.getName(), habitTranslationActual.getName());
+        Assertions.assertEquals(habitTranslation.getDescription(), habitTranslationActual.getDescription());
+        Assertions.assertEquals(habitTranslation.getHabitItem(), habitTranslationActual.getHabitItem());
+        Assertions.assertEquals(habitTranslation.getLanguage(), habitTranslationActual.getLanguage());
+        Assertions.assertEquals(habitTranslation.getHabit(), habitTranslationActual.getHabit());
+
+        Assertions.assertEquals(habitTranslation, habitTranslationActual);
+    }
+
+    @Test
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/HabitTranslationMapperTest.java:47]: Display Name annotation is missing

File: service/src/test/java/greencity/mapping/TagDtoMapperTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import greencity.dto.tag.TagDto;
+import greencity.entity.Tag;
+import greencity.entity.localization.TagTranslation;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class TagDtoMapperTest {
+
+    TagDtoMapper tagDtoMapper = new TagDtoMapper();
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagDtoMapperTest.java:11]: private

File: service/src/test/java/greencity/mapping/TagDtoMapperTest.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            .build();
+
+    TagTranslation target = TagTranslation.builder()
+            .id(5L)
+            .name("Tag")
+            .tag(Tag.builder().id(123L).build())
+            .language(null)
+            .build();
+
+    @Test
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagDtoMapperTest.java:25]: Display Name is missing

File: service/src/test/java/greencity/mapping/TagMapperTest.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import greencity.entity.localization.TagTranslation;
+import greencity.enums.TagType;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TagMapperTest {
+    TagMapper mapper = new TagMapper();
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagMapperTest.java:18]: private

File: service/src/test/java/greencity/mapping/TagMapperTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+            .build();
+
+    final TagVO targetTag = TagVO.builder()
+            .id(1L)
+            .type(TagType.ECO_NEWS)
+            .tagTranslations(List.of(tagTranslationVO))
+            .build();
+
+    @Test
+    void convertTest() {
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagMapperTest.java:50]: Display Name is missing

File: service/src/test/java/greencity/mapping/TagVOMapperTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+import greencity.entity.localization.TagTranslation;
+import greencity.enums.TagType;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TagVOMapperTest {
+    TagVOMapper mapper = new TagVOMapper();
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagVOMapperTest.java:18]: private is missing

File: service/src/test/java/greencity/mapping/TagVOMapperTest.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+            .build();
+
+    final Tag target = Tag.builder()
+            .id(1L)
+            .type(TagType.ECO_NEWS)
+            .tagTranslations(List.of(tt))
+            .build();
+
+    @Test
+    void convertTest() {
```
Comment (nataliiapovzhyk) [inline service/src/test/java/greencity/mapping/TagVOMapperTest.java:50]: Display Name is missing

**Review by nataliiapovzhyk — CHANGES_REQUESTED** [review]
comments added across the code

**Review by df7734 — COMMENTED** [review]
EcoNewsCommentDtoMapperTest.java failed

Expected :EDITED
Actual   :ORIGINAL 

(Last string)

**Comment by nataliiapovzhyk** [comment]
> EcoNewsCommentDtoMapperTest.java failed
> 
> Expected :EDITED Actual :ORIGINAL
> 
> (Last string)

updated ModelUnit to make sure that status is EDITED.

**Comment by nataliiapovzhyk** [comment]
> > EcoNewsCommentDtoMapperTest.java failed
> > Expected :EDITED Actual :ORIGINAL
> > (Last string)
> 
> updated ModelUnit to make sure that status is EDITED.

passes

**Review by nataliiapovzhyk — APPROVED** [review]

**Review by df7734 — APPROVED** [review]

File: service/src/test/java/greencity/mapping/CustomShoppingListMapperTest.java [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+class CustomShoppingListMapperTest {
+
+    @InjectMocks
+    private CustomShoppingListMapper mapper;
+
+    @Test
+    @DisplayName("Test Convert CustomShoppingListItemResponseDto to CustomShoppingListItem")
+    void convert() {
+        CustomShoppingListItemResponseDto toBeConverted = CustomShoppingListItemResponseDto.builder()
+                .id(2L)
```
Comment (BerserkrMM) [inline service/src/test/java/greencity/mapping/CustomShoppingListMapperTest.java:30]: You can use method from ModelUtils to get CustomShoppingListItemResponseDto.

File: service/src/test/java/greencity/mapping/CustomShoppingListResponseDtoMapperTest.java [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+        List<CustomShoppingListItem> listToConvert = new ArrayList<>();
+        listToConvert.add(customShoppingListItem1);
+        listToConvert.add(customShoppingListItem2);
+
+        List<CustomShoppingListItemResponseDto> toVerify = mapper.mapAllToList(listToConvert);
+
+
+        assert ! toVerify.contains(null);
+        assertEquals (2, toVerify.size() );
+        assertEquals(customShoppingListItem1.getText(), toVerify.get(0).getText());
```
Comment (BerserkrMM) [inline service/src/test/java/greencity/mapping/CustomShoppingListResponseDtoMapperTest.java:68]: Item ID has been set but not checked at DTO object.

File: service/src/test/java/greencity/mapping/EcoNewsDtoMapperTest.java [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Test
+    @DisplayName("Test convert form EcoNews to EcoNewsDto")
+    void convert() {
+
+        EcoNews ecoNews = ModelUtils.getEcoNews();
+        EcoNewsDto result = mapper.convert(ecoNews);
+
+        assertNotNull(result);
+        assertEquals(ecoNews.getCreationDate(), result.getCreationDate());
```
Comment (BerserkrMM) [inline service/src/test/java/greencity/mapping/EcoNewsDtoMapperTest.java:28]: ModelUtils.getEcoNews() returned EcoNews with 12 fields. Why you check only 7? (CreationDate,ImagePath,Id,Title,Content,ShortInfo,Author)

Perhaps, if to create expected object, it`ll be more convenient to just compare it with actual result...

File: service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+import greencity.dto.habittranslation.HabitTranslationDto;
+import greencity.entity.HabitTranslation;
+import greencity.entity.Language;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class HabitTranslationDtoMapperTest {
```
Comment (BerserkrMM) [inline service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java]: Public modifiers is not necessary in Tests classes and methods

**Review by BerserkrMM — APPROVED** [review]
All test passes with 100% class coverage

File: service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java (hunk 17 above)
Comment (df7734) [reply service/src/test/java/greencity/mapping/HabitTranslationDtoMapperTest.java]: Fixed. Thanks for review

**Pull request merged**